### ISSUE TITLE
PF3 React: HorizontalNavMenuItem's onItemClick has wrong propType

### ISSUE THREAD

**Issue by martinpovolny** [patternfly/patternfly-react#2994]
Details:

HorizontalNavMenuItem 's onItemClick should be a function not an object.

It's getting passed as `onClick` to `<a href ...>` so it must be a bug.

This causes a React warning each time the onItemClick handler is set.

Fix: https://github.com/patternfly/patternfly-react/pull/2984

**Comment by martinpovolny** [comment]
Illustration of the issue:

```
  console.error node_modules/prop-types/checkPropTypes.js:20
    Warning: Failed prop type: Invalid prop `onItemClick` of type `function` supplied to `HorizontalNavMenuItem`, expected `object`.
        in HorizontalNavMenuItem (created by ToolbarSubmenu)
        in ToolbarSubmenu (created by ToolbarKebab)
        in ul (created by DropdownMenu)
        in RootCloseWrapper (created by DropdownMenu)
        in DropdownMenu (created by Dropdown)
        in div (created by ButtonGroup)
        in ButtonGroup (created by Dropdown)
        in Dropdown (created by Uncontrolled(Dropdown))
        in Uncontrolled(Dropdown) (created by DropdownButton)
        in DropdownButton (created by DropdownButton)
        in DropdownButton (created by ToolbarKebab)
        in div (created by ToolbarKebab)
        in ToolbarKebab (created by ToolbarGroup)
        in span (created by ToolbarGroup)
        in ToolbarGroup (created by Toolbar)
        in div (created by Toolbar)
        in Toolbar (created by WrapperComponent)
        in WrapperComponent
```

**Comment by martinpovolny** [comment]
this is PF3 not PF4

**Comment by tlabaj** [comment]
Closed by #2984
